contracts/sophia.md

```diff
@@ -184,11 +184,45 @@ state associated with each contract instance.
 - State updates are performed by calling a function `put : state => ()`.
 - Aside from the `put` function (and similar functions for transactions
   and events), the language is purely functional.
-- Functions modifying the state need to be annotated with the `stateful` keyword.
+- Functions modifying the state need to be annotated with the `stateful` keyword (see below).
 
 To make it convenient to update parts of a deeply nested state Sophia
 provides special syntax for map/record updates.
 
+#### Stateful functions
+
+Top-level functions (both private and public) must be annotated with the
+`stateful` keyword to be allowed to affect the state of the running contract.
+For instance,
+
+```javascript
+  stateful function set_state(s : state) =
+    put(s)
+```
+
+Without the `stateful` annotation the compiler does not allow the call to
+`put`. A `stateful` annotation is required to
+
+* Use a stateful primitive function. These are
+  - `put`
+  - `Chain.spend`
+  - `Oracle.register`
```

Comment:
Could you please clarify https://github.com/aeternity/protocol/blame/89a8d7511248248b0e42bab3fe2970703fe49b87/contracts/sophia.md#L735-L739

I would expect stateful there? Could you please check the rest of that example?

Reply:
Yeah, those are missing some `stateful` annotations. I'll update.